lib/ansible/plugins/action/reboot.py

```diff
@@ -34,81 +34,130 @@ class ActionModule(ActionBase):
+                                                                                                   module_output['module_stderr'].strip()))
+            except KeyError as ke:
+                if 'failed' in ke.args:
+                    # If there is no failed key, assume the module run worked successfully and set distribution
+                    self._distribution['name'] = module_output['ansible_facts']['ansible_distribution'].lower()
+                    self._distribution['version'] = str(module_output['ansible_facts']['ansible_distribution_version'].split('.')[0])
+                elif 'ansible_distribution' in ke.args:
+                    raise AnsibleError('Failed to find distribution in returned facts.')
+                else:
+                    raise AnsibleError('Failed to find get distribution.')
+        return self._distribution
+
+    @property
+    def shutdown_command(self):
+        if self._shutdown_command is None:
+            shutdown_bin = self.SHUTDOWN_COMMANDS.get(
+                self.distribution['name'],
+                self.DEFAULT_SHUTDOWN_COMMAND)
+
+            find_result = self._execute_module(
+                task_vars=self._task_vars,
+                module_name='find',
+                module_args=dict(
+                    paths=['/sbin', '/usr/sbin', '/usr/local/sbin'],
+                    patterns=[shutdown_bin],
+                    file_type='any'))
+            full_path = [x['path'] for x in find_result['files']]
+            if len(full_path) == 0:
+                raise AnsibleError('Unable to find command "{0}" in system paths.'.format(shutdown_bin))
+            self._shutdown_command = full_path[0]
+        return self._shutdown_command
+
+    @property
+    def shutdown_command_args(self):
+        # Get dist+version specific args first, then distribution args, then use default
+        args = self.SHUTDOWN_COMMAND_ARGS.get(
+            self.distribution['name'] + self.distribution['version'],
+            self.SHUTDOWN_COMMAND_ARGS.get(
+                self.distribution['name'],
+                self.DEFAULT_SHUTDOWN_COMMAND_ARGS))
+        # Convert seconds to minutes. If less that 60, set it to 0.
+        delay_min = self.pre_reboot_delay // 60
+        reboot_message = self._task.args.get('msg', self.DEFAULT_REBOOT_MESSAGE)
+        return args.format(delay_sec=self.pre_reboot_delay, delay_min=delay_min, message=reboot_message)
+
+    @property
+    def reboot_command(self):
+        return '%s %s' % (self.shutdown_command, self.shutdown_command_args)
+
+    @property
+    def boot_time_command(self):
+        return self.BOOT_TIME_COMMANDS.get(self.distribution['name'], self.DEFAULT_BOOT_TIME_COMMAND)
+
+    @property
+    def pre_reboot_delay(self):
+        value = int(self._task.args.get('pre_reboot_delay', self._task.args.get('pre_reboot_delay_sec', self.DEFAULT_PRE_REBOOT_DELAY)))
+        if value < 0:
+            return value
+        else:
+            return value
```

Comment:
I think there's a logic error here.  Perhaps if value < 0: return 0?

Reply:
Yup! Should be `0` if less than `0`.